product_profile/models/product.py

```diff
@@ -0,0 +1,286 @@
+# coding: utf-8
+# © 2015 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from odoo import models, fields, api, _
+from odoo.osv import orm
+from odoo.exceptions import Warning as UserError
+from lxml import etree
+
+
+PROFILE_MENU = (_("Sales > Configuration \n> Product Categories and Attributes"
+                  "\n> Product Profiles"))
+# Prefix name of profile fields setting a default value,
+# not an immutable value according to profile
+PROF_DEFAULT_STR = 'profile_default_'
+LEN_DEF_STR = len(PROF_DEFAULT_STR)
+
+_logger = logging.getLogger(__name__)
+
+
+def format_except_message(error, field, self):
+    value = self.profile_id[field]
+    model = type(self)._name
+    message = (_("Issue\n------\n"
+                 "%s\n'%s' value can't be applied to '%s' field."
+                 "\nThere is no matching value between 'Product Profiles' "
+                 "\nand '%s' models for this field.\n\n"
+                 "Resolution\n----------\n"
+                 "Check your settings on Profile model:\n%s"
+               % (error, value, field, model, PROFILE_MENU)))
+    return message
+
+
+def get_profile_fields_to_exclude():
+    # These fields must not be synchronized between product.profile
+    # and product.template/product
+    return models.MAGIC_COLUMNS + [
+        'name', 'explanation', 'sequence',
+        'display_name', '__last_update']
+
+
+class ProductProfile(models.Model):
+    _name = 'product.profile'
+    _order = 'sequence'
+
+    name = fields.Char(
+        required=True,
+        help="Profile name displayed on product template\n"
+             "(not synchronized with product.template fields)")
+    sequence = fields.Integer(
+        help="Defines the order of the entries of profile_id field\n"
+             "(not synchronized with product.template fields)")
+    explanation = fields.Text(
+        required=True,
+        oldname='description',
+        help="An explanation on the selected profile\n"
+             "(not synchronized with product.template fields)")
+    type = fields.Selection(
+        selection='_get_types',
```

Comment:
New API provides a `selection_add` attribute, so you don't need to use a method to allow other modules to extend the selection list. You can directly put `selection=[('consu', 'Consumable'), ('service', 'Service')]` here instead.